plugins/externalwrench/src/ApplyExternalWrench.cc

```diff
@@ -310,19 +115,46 @@ void RPCServerThread::run()
             this->m_reply.addString ( "[duration]: (double) Duration of the applied force in seconds" );
             this->m_reply.addString ( "Note: The reference frame is the base/root robot frame with x pointing backwards and z upwards.");
             this->m_rpcPort.reply ( this->m_reply );
-        } else {
-            if ( command.get ( 0 ).isString() \
+        } else{
+            if(command.get(0).isString() \
                     && ( command.get ( 1 ).isDouble() || command.get ( 1 ).isInt() )  && ( command.get ( 2 ).isDouble() || command.get ( 2 ).isInt() ) && ( command.get ( 3 ).isDouble() || command.get ( 3 ).isInt() ) \
                     && ( command.get ( 4 ).isDouble() || command.get ( 4 ).isInt() )  && ( command.get ( 5 ).isDouble() || command.get ( 5 ).isInt() ) && ( command.get ( 6 ).isDouble() || command.get ( 6 ).isInt() )  \
                     && ( command.get ( 7 ).isDouble() || command.get ( 7 ).isInt() ) ) {
                 this->m_reply.addString ( "[ACK] Correct command format" );
                 this->m_rpcPort.reply ( m_reply );
                 m_lock.lock();
                 // new-command flag
-                command.addInt32(1);
+                command.addInt(1);
                 m_cmd = command;
                 m_lock.unlock();
-            } else {
+
+                if (this->m_mode == "single") {
+
+                    // Delete the previous wrenchs
+                    if (wrenchesVectorPtr->size() != 0) {
+                        boost::shared_ptr<ExternalWrench> wrench = wrenchesVectorPtr->at(0);
+                        wrench->deleteWrench();
+                        wrenchesVectorPtr->clear();
+                    }
+
+                }
+
+                //Creating new instances of external wrenches
+                boost::shared_ptr<ExternalWrench> newWrench(new ExternalWrench);
+                if(newWrench->setWrench(m_robotModel, m_cmd))
+                {
+                    wrenchesVectorPtr->push_back(newWrench);
+                }
+                else yError() << "Failed to set new wrench values!";
```

Comment:
Here it should be possible to reply with an `rpc` error message as [done below](https://github.com/robotology/gazebo-yarp-plugins/pull/418/files#diff-6f34d47dcd5558cbef158050b5563c76R158):
```
this->m_reply.clear();
this->m_reply.addString ( "ERROR: unable to get the desired link" );
this->m_rpcPort.reply ( this->m_reply );
```

Reply:
Updated the response handling